src/ContainerInterface.php

```diff
@@ -6,6 +6,8 @@
 
 /**
  * Describes the interface of a container that exposes methods to read its entries.
+ *
+ * @template T of mixed
```

Comment:
This template doesn't make sense: a container could hold any number of types, what's the reason for declaring such `T`, without linking it to anything?

Reply:
The reason for declaring `T` as mixed is to keep the interface as generic as possible so that implementers can decide if they want to keep it generic or declare a specific typehint. 

I've updated the PR description with slightly better explanation and example code.